### PR TITLE
Feature/implement disenchant loot

### PR DESCRIPTION
**Pull request by xadamxk**
I need to add a second prompt for disenchantments but I thought I'd make this PR early incase you had time.

### CODE REVIEW

**Comment by xadamxk** [comment]
@TyrantKingBen This branch should be ready for review if you get a chance to do so this week.

File: src/electron/enums/index.ts [hunk 1]
```diff
@@ -1,3 +1,7 @@
 export * from './endpoints';
 export * from './gameflow-phase';
 export * from './player-response';
+export * from './loot-categories';
```
Comment (xadamxk) [inline src/electron/enums/index.ts:3]: @TyrantKingBen Should I move loot into a child folder and just import everything in that folder?

File: src/electron/modules/disenchant-loot.ts [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+    const menuItem = new MenuItem({
+      label: 'Disenchant Loot',
+      sublabel: '',
+      submenu: submenu
+    });
+
+    this.updateMenu(menuItem);
+
+    submenu.append(new MenuItem({
```
Comment (xadamxk) [inline src/electron/modules/disenchant-loot.ts:18]: I'll turn this into an object and loop through it to add menu items. Just realized how much code is repeated.

**Comment by TyrantKingBen** [comment]
I still need to look at the models/responses involved, but functionally it looks all to be working without any potential bugs now. The message boxes look a little weird to me.

**Comment by TyrantKingBen** [comment]
When you pull the commits I added, make sure you do `npm i` to install the newer version ESLint packages.

**Review by xadamxk — COMMENTED** [review]
> I still need to look at the models/responses involved, but functionally it looks all to be working without any potential bugs now. The message boxes look a little weird to me.

I had tested every disenchant option. I wanted to disable the buttons until the checkbox is checked, but I'm not sure it is possible with Electron.

File: src/electron/enums/index.ts (hunk 1 above)
Comment (TyrantKingBen) [reply src/electron/enums/index.ts:3]: Perhaps. I need to look at the responses from the API to see what makes the most sense.

**Comment by TyrantKingBen** [comment]
> > I still need to look at the models/responses involved, but functionally it looks all to be working without any potential bugs now. The message boxes look a little weird to me.
> 
> I had tested every disenchant option. I wanted to disable the buttons until the checkbox is checked, but I'm not sure it is possible with Electron.

Yeah, I need to look at the dialog API some more to see what’s actually possible.

**Pull request merged**